.github/workflows/cleanup.yml

```diff
@@ -67,3 +64,66 @@ jobs:
               echo "Deleting container group: ${container}"
               az resource delete --ids ${container}
           done
+
+  cleanup-hsm:
+    name: Cleanup HSM Keys
+    runs-on: ubuntu-latest
+    steps:
+      - name: Log into Azure
+        uses: azure/login@v2
+        with:
+          client-id: ${{ secrets.MANAGED_ID_CLIENT_ID }}
+          tenant-id: ${{ secrets.MANAGED_ID_TENANT_ID }}
+          subscription-id: ${{ vars.SUBSCRIPTION }}
+
+      - name: Delete HSM Keys
+        run: |
+          az keyvault key list --hsm-name ${{ vars.HSM_NAME }} \
+            --query "[].kid" -o tsv | \
+            xargs -I{} az keyvault key delete --id {}
+
+  cleanup_registry:
+    name: Cleanup Container Registry
+    runs-on: ubuntu-latest
+    steps:
+      - name: Log into Azure
+        uses: azure/login@v2
+        with:
+          client-id: ${{ secrets.MANAGED_ID_CLIENT_ID }}
+          tenant-id: ${{ secrets.MANAGED_ID_TENANT_ID }}
+          subscription-id: ${{ vars.SUBSCRIPTION }}
+
+      - name: Log in to Azure Container Registry
+        run: az acr login --name ${{ secrets.REGISTRY_NAME }}
+
+      - name: Cleanup Registry
+        run: |
+          # The filter looks for at least one digit as a proxy for auto
+          # generated tags
+          az acr run \
+            --cmd "acr purge --filter '.*:.*\d+.*' --ago 1d" \
+            --registry ${{ secrets.REGISTRY_NAME }} \
+            /dev/null
+
+  cleanup-storage-account:
+    name: Cleanup Storage Account Blobs
+    runs-on: ubuntu-latest
+    steps:
+      - name: Log into Azure
+        uses: azure/login@v2
+        with:
+          client-id: ${{ secrets.MANAGED_ID_CLIENT_ID }}
+          tenant-id: ${{ secrets.MANAGED_ID_TENANT_ID }}
+          subscription-id: ${{ vars.SUBSCRIPTION }}
+
+      - name: Delete Blobs
+        run: |
+          az storage blob list \
```

Comment:
Up to you, I think it's a best effort to stop things like storage accounts getting too full, if it fails as a one off I don't think it matters too much, and if it started failing consistently we would see the job history and take more action

Reply:
Gotcha, makes sense. I think it'd be helpful to be able to easily tie the resources back to the test run for RCA if something like consistent failure were to happen. Is there already a way to do that currently? If not, logs or just adding a tag to the resource through the bicep file ([example](https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/tag-resources-bicep#apply-values)) could work

Reply:
Yeah I can add a tag that a given resource came from this repo if you'd like? They all go into the same `confidential-sidecars` resource group so I think that would make tracing easy enough but happy to take input on that

Reply:
Gotcha, I was thinking of a direct tie between the GitHub run number/id and the resources created. Like you mentioned, we would already know the that they came from this repo since they all go into the same resource group. But I think having a way to relate resource to run would be convenient. For example, suppose you find a resource in the resource group (so it wasn't cleaned up). Having a tag on the resource specifying the run number/id would make it easy to see what runs to lookup.

Reply:
Makes sense, for that use case, the container groups are named based on the run id and run number so that can be used to relate back to the run